core/src/main/scala/kafka/log/Log.scala

```diff
@@ -959,10 +968,14 @@ object Log {
    * Parse the topic and partition out of the directory name of a log
    */
   def parseTopicPartitionName(dir: File): TopicAndPartition = {
-    val name: String = dir.getName
+    var name: String = dir.getName
```

Comment:
Can you restructure this to use `val` - it helps to have a single block on the RHS that encapsulates the full assignment logic (as opposed to being exposed to the method's entire scope).